packages/ui/slide-over.tsx

```diff
@@ -0,0 +1,162 @@
+        onClose={() => setIsOpen(false)}
+      >
+        <div className="fixed inset-0" />
+
+        <div className="fixed inset-0 overflow-hidden">
+          <div className="absolute inset-0 overflow-hidden">
+            <div className="pointer-events-none fixed inset-y-0 right-0 flex max-w-full pl-10">
+              <Transition.Child
+                as={Fragment}
+                enter="transform transition ease-in-out duration-500 sm:duration-700"
+                enterFrom="translate-x-full"
+                enterTo="translate-x-0"
+                leave="transform transition ease-in-out duration-500 sm:duration-700"
+                leaveFrom="translate-x-0"
+                leaveTo="translate-x-full"
+              >
+                <Dialog.Panel className={panelClassName}>
+                  <div className="flex h-full flex-col shadow-xl">
+                    {children}
+                  </div>
+                </Dialog.Panel>
+              </Transition.Child>
+            </div>
+          </div>
+        </div>
+      </Dialog>
+    </Transition.Root>
+  );
+}
+
+interface OpenButtonProps {
+  children: ReactNode;
+  className?: string;
+}
+
+function SlideOverOpenButton({children, className}: OpenButtonProps) {
+  const {setIsOpen} = useContext(SlideOverContext);
+
+  return (
+    <button className={className} onClick={() => setIsOpen(isOpen => !isOpen)}>
+      {children}
+    </button>
+  );
+}
+
+interface SlideOverHeaderProps {
+  children?: ReactNode;
+}
+
+function SlideOverHeader({children}: SlideOverHeaderProps = {}) {
+  const {variant, setIsOpen} = useContext(SlideOverContext);
+
+  const headerClassName = classNames({
+    'p-4 sm:px-6': variant === 'text',
+    'bg-neutral-700': variant === 'gallery',
+  });
+  const buttonClassName = classNames(
+    'rounded-md hover:text-gray-500 focus:outline-none focus:ring-2 focus:ring-indigo-500 focus:ring-offset-2',
+    {
+      'bg-white text-gray-400': variant === 'text',
+      'bg-black text-white p-2 my-2 mx-6': variant === 'gallery',
+    }
+  );
+
+  return (
+    <div className={headerClassName}>
+      <div className="flex items-center justify-end">
+        {children}
+        <div className="ml-3 flex items-center">
+          <button
+            type="button"
+            className={buttonClassName}
+            onClick={() => setIsOpen(false)}
+          >
+            <span className="sr-only">Close panel</span>
+            <XMarkIcon className="h-6 w-6" aria-hidden="true" />
+          </button>
+        </div>
+      </div>
+    </div>
+  );
+}
+
+function SlideOverContent({children}: SlideOverDialogProps) {
+  const {variant} = useContext(SlideOverContext);
+  const className = classNames('relative flex-1', {
+    'mt-6 px-4 sm:px-6': variant === 'text',
+  });
+
+  return <div className={className}>{children}</div>;
+}
+
+export {
+  SlideOver,
+  SlideOverOpenButton,
+  SlideOverDialog,
+  SlideOverHeader,
+  SlideOverContent,
```

Comment:
Would it be possible to make the `export` a part of the definition of each function (`export function SlideOverContent...`)?